File: output/erlang_guard_test.rb
def age(ageinput)
  case
  when ageinput>19
    :adult
  when ageinput>=13; ageinput<=19
    :teen
  when ageinput>=3; ageinput<13
    :child
  when ageinput>=1; ageinput<3
    :toddler
  end
end

puts("Enter some age: ")
ageinput = gets.chomp
puts("Result: #{age(ageinput)}")

File: transpiler/erlang_guard_test.erl
-module(erlang_guard_test).
-compile(export_all).

age(AgeInput) when AgeInput > 19 ->
  adult;
age(AgeInput) when AgeInput >= 13, AgeInput =< 19 ->
  teen;
age(AgeInput) when AgeInput >= 3, AgeInput < 13 ->
  child;
age(AgeInput) when AgeInput >= 1, AgeInput < 3 ->
  toddler.

main() ->
  AgeInput = 20,
  io:format("Result: ~w~n", [age(AgeInput)]).
